sagemaker/launch/launch_realtime_endpoint.py

```diff
@@ -0,0 +1,269 @@
+
+    logging.info("Waiting for {} endpoint to be in service...".format(sm_ep_name))
+    waiter = sm_client.get_waiter("endpoint_in_service")
+    
+    try:
+        waiter.wait(EndpointName=sm_ep_name,
+                    WaiterConfig={
+                        'Delay': 30,        # seconds between querying
+                        'MaxAttempts': 60   # max retries (~30 minutes here)
+                    })
+        logging.info('{} endpoint has been successfully created, and ready to be \
+                 invoked!'.format(sm_ep_name))
+    except WaiterError as e:
+        logging.error("Waiter timed out or endpoint creation failed:", e)
+
+
+def get_realtime_infer_parser():
+    """
+    Get GraphStorm realtime  task parser.
+    """
+    realtime_infer_parser = argparse.ArgumentParser("GraphStorm Inference Args")
+
+    # SageMaker specific arguments
+    realtime_infer_parser.add_argument("--image-url", type=str, required=True,
+        help="GraphStorm SageMaker docker image URI")
+    realtime_infer_parser.add_argument("--role", type=str, required=True,
+        help="SageMaker execution role")
+    realtime_infer_parser.add_argument("--region", type=str, required=True,
+        help="AWS region to launch jobs in. Make sure this region is where the inference image, \
+             and model tar file are located!")
+    realtime_infer_parser.add_argument("--instance-type", type=str, default="ml.c6i.xlarge",
+        help="instance type for the SageMaker job")
+    realtime_infer_parser.add_argument("--instance-count", type=int, default=1,
+        help="number of instances")
+
+    # real-time task specific arguments
+    realtime_infer_parser.add_argument("--restore-model-path", type=str,
+        help="The folder path where trained GraphStorm model parameters were saved.")
+    realtime_infer_parser.add_argument("--model-yaml-config-file", type=str,
+        help="The file path to the new YAML configuration file generated in GraphStorm \
+              model training.")
+    realtime_infer_parser.add_argument("--graph-json-config-file", type=str,
+        help="The file path to the updated JSON configuration file created in GraphStorm \
+              graph construction process.")
+    realtime_infer_parser.add_argument("--upload-tarfile-s3", type=str,
+        help="The S3 location for uploading the packed model artifacts tar file.")
+    realtime_infer_parser.add_argument("--infer-task-type", type=str,
+        choices=SUPPORTED_REALTIME_INFER_TASKS,
+        help=(f"The name of real time inference task. Options \
+               include {SUPPORTED_REALTIME_INFER_TASKS}"))
+    realtime_infer_parser.add_argument("--model-name", type=check_name_format,
+        default='GSF-Model4Realtime',
+        help="The name for the to-be created SageMaker objects. The name should follow \
+              a regular expression pattern: ^[a-zA-Z0-9]([\-a-zA-Z0-9]*[a-zA-Z0-9])$. Default \
+              is \"GSF-Model4Realtime\".")
```

Comment:
The default name does not follow the regular expression pattern.

Reply:
It is.